src/gapp_login/wechat/endpoints.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from urllib import parse
+
+from authlib_gino.fastapi_session.api import auth, login_context
+from authlib_gino.fastapi_session.models import User
+from fastapi import APIRouter, HTTPException, Form, FastAPI, Depends
+from starlette import status
+from starlette.requests import Request
+
+from . import config
+from .models import db, WeChatIdentity
+from .utils import WeChatAuthError
+
+log = logging.getLogger(__name__)
+router = APIRouter()
+
+
+@router.put("/login/wechat")
+async def login_wechat(
+    request: Request, code: str = Form(...), ctx=Depends(login_context)
+):
+    idp = ctx["idp"]
+    idp_params = dict(parse.parse_qsl(parse.unquote(ctx["idp_params"])))
+    appid = idp_params.get("appid")
+    if not appid or appid not in config.WECHAT_CLIENTS:
+        raise HTTPException(status.HTTP_403_FORBIDDEN, "Wrong wechat AppID")
+
+    async with config.WECHAT_CLIENTS[appid] as client:
```

Comment:
In order to properly share the underlying `AsyncClient`, we only need to `aclose()` the `AsyncClient` instances before the application stops for just once, and remove this `async with` here. (`AsyncClient.__aenter__` is not doing anything as you noticed that too)

Reply:
fixed